src/src/Controller/AdController.php

```diff
@@ -22,6 +22,17 @@ public function index(AdRepository $adRepository): Response
         ]);
     }
 
+    #[Route('/popular', name: 'app_ad_popular')]
+    public function mostPopular(AdRepository $adRepository, Request $request)
+    {
+        $search = $request->request->get('q');
+        $popularAds = $adRepository->findMostPopular($search);
+
+        return $this->render('ad/popular.html.twig', [
+            'ads' => $popularAds
+        ]);
+    }
+
```

Comment:
When we search we need to go on `/ad` route with our ad filtered (I think search query can be passed in url)